### PR TITLE
bug: issue-4625 repro

### PR DESCRIPTION
**Pull request by crusso**
Non-dfx repro for issue #4625

### CODE REVIEW

File: test/run-drun/issue-4625/call.mo [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import lib "lib";
+
+actor {
+    public shared func f(x: lib.X): async () {
```
Comment (ggreif) [inline test/run-drun/issue-4625/call.mo:4]:
```suggestion
    public shared func f(x : lib.X) : async () {
```

File: test/run-drun/issue-4625.mo [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//MOC-FLAG --actor-alias call lg264-qjkae
+import lib "issue-4625/lib";
+import call "canister:call";
+
+actor {
+  type T = actor {
+    f(x: lib.X): async ();
+  };
+
+  let z: T = call;
```
Comment (ggreif) [inline test/run-drun/issue-4625.mo:11]:
```suggestion
    f(x : lib.X) : async ();
  };

  let z : T = call;
```

**Review by ggreif — DISMISSED** [review]
LGTM, maybe you can add the 3 missing newlines too.

File: test/run-drun/issue-4625.mo [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    f(x : lib.X) : async ();
+  };
+
+  let z : T = call;
+};
+
+//SKIP run
+//SKIP run-ir
+//SKIP run-low
+//SKIP ic-ref
```
Comment (crusso) [inline test/run-drun/issue-4625.mo:17]:
```suggestion
//SKIP ic-ref

```

File: test/run-drun/issue-4625/call.mo [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import lib "lib";
+
+actor {
+    public shared func f(x : lib.X) : async () {
+        ();
+    };
+}
```
Comment (crusso) [inline test/run-drun/issue-4625/call.mo:7]:
```suggestion
}

```

**Comment by ggreif** [comment]
Maybe this should be under `test/fail`, then?

**Comment by crusso** [comment]
I'm not sure how fixable this is.  I wonder if design/IDL-Motoko.md is underspecified on how to translate Motoko/Candid references to type definitions, i.e. translate without or after expansion and that affects the translation of argument sequences.



In addition, both Motoko Null and () map to the Candid null.

so type X = () translates to type X = Null
but
func f(x : X) : async X has Candid type () -> ()
 (I think), not, null -> null.

**Pull request closed without merging**